packages/react/src/components/DataTable/state/getDerivedStateFromProps.js

```diff
@@ -37,12 +38,12 @@ const getDerivedStateFromProps = (props, prevState) => {
     shouldShowBatchActions: prevState.shouldShowBatchActions || false,
   };
 
-  if (prevState.sortDirection && prevState.sortHeaderKey) {
+  if (state.sortDirection && state.sortHeaderKey) {
```

Comment:
I'm assuming `prevState` was being used previously to prevent `getSortedState` from being called when it doesn't need to (when there's no prevState and `sortDirection` = `initialSortState` and `sortHeaderKey` = `null`). Would this be a better condition?

```suggestion
  if ((prevState.sortDirection && prevState.sortHeaderKey) || (props.initialSortDirection && props.initialSortcolumn )) {
```

Or can you think of another option that would continue to ensure `getSortedState` isn't called when it previously wasn't called?